flow/raster_cache.cc

```diff
@@ -34,8 +34,10 @@ void RasterCacheResult::draw(SkCanvas& canvas, const SkPaint* paint) const {
 
   SkRect bounds =
       RasterCacheUtil::GetDeviceBounds(logical_rect_, canvas.getTotalMatrix());
-  FML_DCHECK(std::abs(bounds.width() - image_->dimensions().width()) <= 1 &&
-             std::abs(bounds.height() - image_->dimensions().height()) <= 1);
+  // clang-format off
+  FML_DCHECK(std::round(std::abs(bounds.width() - image_->dimensions().width())) <=1 &&
```

Comment:
std::round is not the answer. If this delta is >1 then we have a problem somewhere. This DCHECK is there for a reason.

Now, round-off error in the range of .00012 likely doesn't represent a big problem and we could probably just accept it, though it would be interesting to see why our math allowed that round-off error to affect the calculations here in the first place.

If we are going to fix this bug by relaxing the assertion, then we should use 1+epsilon rather than round which will allow up to 1+0.5. One concept for an acceptable epsilon is whether or not it would affect the next pixel which would require a coverage of at least 1/256 (or 1/512 representing half of an AA sample) since that would be the smallest coverage that might cause that pixel to rise from transparency to a tiny nearly invisible pixel.

But, I'd rather see an analysis of where we incurred the round-off.

Also, I'm not sure about the abs() function. The image dimensions should always be larger than the device bounds, I'd rather see the subtraction reversed and >=0 <=1 be used as the test.